src/MICore/Transports/PipeTransport.cs

```diff
@@ -137,6 +137,11 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             proc.StartInfo.Arguments = pipeOptions.PipeArguments;
             proc.StartInfo.WorkingDirectory = pipeOptions.PipeCwd;
 
+            foreach (EnvironmentEntry entry in pipeOptions.PipeEnvironment)
+            {
+                proc.StartInfo.SetEnvironmentVariable(entry.Name, entry.Value);
```

Comment:
I see StartInfo.SetEnvironmentVariable is used in LocalTransport.cs as well. 
@andrewcrawley @faxue-msft  can you please confirm?